src/index.ts

```diff
@@ -1,66 +1,71 @@
 class School {
-	directions = [];
+	directions: Direction[] = [];
  
-	addDirection(direction) {
+	addDirection(direction: Direction): void {
 	  this.directions.push(direction);
 	}
  }
  
  class Direction {
-	levels = [];
+	levels: Level[] = [];
+	_name: string;
  
-	get name() {
+	get name(): string {
 	  return this._name;
 	}
  
-	constructor(name) {
+	constructor(name: string) {
 	  this._name = name;
 	}
  
-	addLevel(level) {
+	addLevel(level: Level): void {
 	  this.levels.push(level);
 	}
  }
  
  class Level {
-	groups = [];
+	groups: Group[] = [];
+	_name: string;
+	_program: string;
  
-	constructor(name, program) {
-	  this.name = name;
+	constructor(name: string, program: string) {
+	  this._name = name;
 	  this._program = program;
 	}
  
-	get name() {
+	get name(): string {
 	  return this._name;
 	}
  
-	get program() {
+	get program(): string {
 	  return this._program;
 	}
  
-	addGroup(group) {
+	addGroup(group: Group): void {
 	  this.groups.push(group);
 	}
  }
  
  class Group {
-	_students = [];
+	_students: Student[] = [];
+	directionName: string;
+	levelName: string;
  
-	get students() {
+	get students(): Student[] {
 	  return this._students;
 	}
  
-	constructor(directionName, levelName) {
+	constructor(directionName: string, levelName: string) {
 	  this.directionName = directionName;
 	  this.levelName = levelName;
 	}
  
-	addStudent(student) {
+	addStudent(student: Student): void {
 	  this._students.push(student);
 	}
  
-	showPerformance() {
-	  const sortedStudents = this.students.toSorted(
+	showPerformance(): Student[] {
+	  const sortedStudents = this._students.slice().sort(
```

Comment:
Don't forget to add type annotations for your variables and constants. It can be very helpful, and it's a good opportunity to practice.